pkg/controller/core/clusterqueue_controller.go

```diff
@@ -67,17 +67,20 @@ func (r *ClusterQueue) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Re
 	ctx = ctrl.LoggerInto(ctx, log)
 	log.V(2).Info("Reconciling ClusterQueue")
 
+	// Shallow copy enough for now.
+	oldStatus := cqObj.Status
+
 	usage, workloads, err := r.cache.Usage(&cqObj)
 	if err != nil {
 		log.Error(err, "Failed getting usage from cache")
 		// This is likely because the cluster queue was recently removed,
 		// but we didn't process that event yet.
 		return ctrl.Result{}, err
 	}
-	// Shallow copy enough for now.
-	oldStatus := cqObj.Status
+
```

Comment:
why this move? seems better to have this defined as close as possible to where it is used.

Reply:
I can move it back, during devel, it sounded better to be there, but maybe just to my eyes

Reply:
actually, we don't need this copy, it seems we are recreating the whole status every time. I suggest we create a function Status(), something like this:

```
func (r *ClusterQueue) Status(cq *kueue.ClusterQueue) (*kueue.ClusterQueueStatus, error) {
   usage, workloads, err := r.cache.Usage(&cqObj)
   if err != nil {
	log.Error(err, "Failed getting usage from cache")
	// This is likely because the cluster queue was recently removed,
	// but we didn't process that event yet.
	return nil, err
   }
   return &kueue.ClusterQueueStatus{
      UsedResources: usage,
      AdmittedWorkloads: int32(workloads),
      PendingWorkloads: r.qManager.Pending(cqj)
   }, nil
}
```

Reply:
we still need the old one for 
```go
!equality.Semantic.DeepEqual(oldStatus, cqObj.Status)
```

Reply:
I like the wrapper idea, but we need the old one to avoid updating with the same info

Reply:
unless we add a timestamp

Reply:
```go
func (r *ClusterQueue) Status(cq *kueue.ClusterQueue) (*kueue.ClusterQueueStatus, error) {
	usage, workloads, err := r.cache.Usage(cq)
	now := metav1.Now()
	if err != nil {
		r.log.Error(err, "Failed getting usage from cache")
		// This is likely because the cluster queue was recently removed,
		// but we didn't process that event yet.
		return nil, err
	}
	return &kueue.ClusterQueueStatus{
		UsedResources:     usage,
		AdmittedWorkloads: int32(workloads),
		PendingWorkloads:  r.qManager.Pending(cq),
		LastUpdateTime:    now,
	}, nil
}
```
Would look like this

Reply:
why do we need the timestamp? can't we do something like this (give the Status func above):

```
status, err := r.Status(cqObj)
if err != nil {
  return ctrl.Result{}, err
}
if !equality.Semantic.DeepEqual(status, cqObj.Status) {
  cqObj.Status = status
  err = r.client.Status().Update(ctx, &cqObj)
  return ctrl.Result{}, client.IgnoreNotFound(err)
}

```

Reply:
done!